### DISCUSSION TITLE
How to add a base header

### DISCUSSION THREAD

**Discussion by booozz** [flutterdata/flutter_data#139] [Q&A] [answered]
Hi all,

first of all my english is not the best but i hope that you understand what i mean if not feel free to ask!

I'm trying this plugin for my app and i have a server that uses users for the note's and a header for authentication.
now i was wondering if you can add a base header for your request like get al the note's.

i have lookt in the docs but thear you can add a header by a custom request(adapter),
but noting that says you can add a header by the baseurl or somting like that.

is it posible? or can anyone give me some extra information?

Thanks by Boaz!

**Comment by oarshad** [comment, marked as answer]
You can set that in your adapter like so:

```
mixin ApplicationAdapter<T extends DataModel<T>> on RemoteAdapter<T> {
  @override
  String get baseUrl => "http://localhost:8080/api/v1/";

  @override
  FutureOr<Map<String, String>> get defaultHeaders async {
    final token = await read(tokenProvider.future);
    if (token == null) {
      return await super.defaultHeaders;
    } else {
      return await super.defaultHeaders
        ..addAll({'Authorization': 'Bearer $token'});
    }
  }
}
```
